Mathlib/GroupTheory/Coxeter/StandardGeometricRepresentation.lean

```diff
@@ -0,0 +1,285 @@
+    simp only [← add_assoc, h₁, h₂, h₃, h₄, h₅]
+    clear h₁ h₂ h₃ h₄ h₅
+    -- Simplify using the sine and cosine angle addition formula.
+    have h₆ : ((2 * (Nat.succ k) + 1) * (π / m)) = 2 * k * (π / m) + π / m + π / m + π / m := by
+      rw [Nat.succ_eq_add_one]
+      push_cast
+      ring
+    have h₇ : ((2 * (Nat.succ k)) * (π / m)) = 2 * k * (π / m) + π / m + π / m := by
+      rw [Nat.succ_eq_add_one]
+      push_cast
+      ring
+    have h₈ : ((2 * k + 1) * (π / m)) = 2 * k * (π / m) + π / m := by ring
+    simp only [h₆, h₇, h₈, sin_add, cos_add]
+    clear h₆ h₇ h₈
+    -- Now equate the coefficients of `v` and `v'`.
+    congr
+    · field_simp [sin_pi_div_m_ne_zero hm]
+      have := sin_sq_add_cos_sq (π / m)
+      linear_combination
+        (3 * sin (2 * k * π / m) * cos (π / m) + cos (2 * k * π / m) * sin (π / m)) * this
+    · field_simp [sin_pi_div_m_ne_zero hm]
+      have := sin_sq_add_cos_sq (π / m)
+      linear_combination sin (2 * k * π / m) * this
+
+private lemma orthoReflection_mul_orthoReflection_pow_order_apply_v {v v' : V} {m : ℕ} (hm : 1 < m)
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) :
+    (((r hv) * (r hv')) ^ m) v = v := by
+  rw [M.orthoReflection_mul_orthoReflection_pow_apply m hm hv hv' hvv']
+  rw [add_mul, mul_assoc 2, mul_div_cancel₀ _ (by positivity)]
+  simp [add_comm, sin_add_two_pi, sin_two_pi, div_self (sin_pi_div_m_ne_zero hm)]
+
+private lemma orthoReflection_mul_orthoReflection_pow_order_apply_v' {v v' : V} {m : ℕ} (hm : 1 < m)
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) :
+    (((r hv) * (r hv')) ^ m) v' = v' := let a := r hv; let b := r hv'; calc
+  ((a * b) ^ m) v'
+  _ = (b * b * (a * b) ^ m) v'         := by simp [M.orthoReflection_sq hv']
+  _ = (b * (b * (a * b) ^ m)) v'       := by rw [mul_assoc]
+  _ = (b * ((b * a) ^ m * b)) v'       := by
+    congr 2
+    exact (SemiconjBy.eq (SemiconjBy.pow_right (by unfold SemiconjBy; group) m))
+  _ = (b * (b * a) ^ m * b) v'         := by rw [mul_assoc]
+  _ = (b * (b * a) ^ m) (b v')         := LinearMap.mul_apply _ _ _
+  _ = (b * (b * a) ^ m) (-v')          := congrArg _ (M.orthoReflection_apply_self hv')
+  _ = -((b * (b * a) ^ m) v')          := map_neg _ _
+  _ = -(b (((b * a) ^ m) v'))          := congrArg _ (LinearMap.mul_apply _ _ _)
+  _ = -(b v')                          := by
+    congr
+    apply M.orthoReflection_mul_orthoReflection_pow_order_apply_v hm hv' hv
+    · rwa [← M.standardBilinForm_comm v v']
+  _ = -(-v')                           := congrArg _ (M.orthoReflection_apply_self hv')
+  _ = v'                               := neg_neg v'
+
+private lemma can_decomp_into_parallel_and_orthogonal {v v' : V} (w : V) {m : ℕ}
+    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) (hm : m > 1) :
+    ∃ μ₁ μ₂ : ℝ, ⟪v, w - μ₁ • v - μ₂ • v'⟫ = 0 ∧ ⟪v', w - μ₁ • v - μ₂ • v'⟫ = 0 := by
```

Comment:
The following phrasing of this lemma may be easier to use:
```suggestion
private lemma can_decomp_into_parallel_and_orthogonal {v v' : V} (w : V) {m : ℕ}
    (hv : ⟪v, v⟫ = 1) (hv' : ⟪v', v'⟫ = 1) (hvv' : ⟪v, v'⟫ = -cos (π / m)) (hm : m > 1) :
    ∃ (w' : V) (μ₁ μ₂ : ℝ) (hw : w = w' + μ₁ • v + μ₂ • v'),
      ⟪v, w'⟫ = 0 ∧ ⟪v', w'⟫ = 0 := by
```
Then, you may use the following syntax in the proof below:
```lean
    obtain ⟨w', μ₁, μ₂, rfl, h₁, h₂⟩ := M.can_decomp_into_parallel_and_orthogonal w hv hv' hvv' mgt
```